### PR TITLE
form: improve error message

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds catch error on record service on create, update and delete functions.
* Adds a new exception for the editor.
* Adds error message on the form.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 1]
```diff
@@ -913,4 +916,14 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
       };
     }
   }
+
+  /**
+   * Handle form error
+   * @return Observable<Error>
+   */
+  private _handleError(error: { status: BigInteger, title: string }): Observable<Error> {
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:924]: Why `BigInteger`, this cannot be just `number`?

**Review by sebastiendeleze — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/editor/editor.component.ts:924]: done

File: projects/rero/ng-core/src/lib/record/record.service.ts [hunk 2]
```diff
@@ -153,8 +153,8 @@ export class RecordService {
     return this._http
       .delete<void>(this._apiService.getEndpointByType(type, true) + '/' + pid)
       .pipe(
-        tap(() => this.onDelete.next(this.createEvent(type, { pid }))),
-        catchError((error) => this.handleError(error))
+        catchError((error) => this.handleError(error)),
+        tap(() => this.onDelete.next(this.createEvent(type, { pid })))
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/record.service.ts:157]: `createEvent` is private you can prefix with `_`

**Review by jma — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/record.service.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/record.service.ts]: done

**Pull request merged**